AddressSpace/CMakeLists.txt

```diff
@@ -16,7 +16,12 @@
 
 # Author: Piotr Nikiel <piotr@nikiel.info>
 
-include(${PROJECT_BINARY_DIR}/AddressSpace/cmake_generated.cmake)
+# QUASAR_PURE_CMAKE_BUILD should evaluate to true whenever plain CMake is the entry point, like in Yocto
+if(${QUASAR_PURE_CMAKE_BUILD})
```

Comment:
No, it is very weakly related to in-source/out-of-source (FYI, to determine in/out-of-source you compare native CMake variables like and you don't need to supply a custom one).

The motivation of this switch is that certain build environments will integrate a quasar project on a level of CMake inclusion (including project's master CMakeLists.txt) rather than via using our build command (./quasar.py build).
If that happens then so called "cmake headers" (essentially, a Design-dependent list of classes) will not be generated by "quasar.py build" (because "quasar.py build" will never be called) and needs to be pre-generated by a developer. In such a case it is treated as a source-code (and stored in the repo) rather than a build artifact. Therefore, it should be taken from source directory and not the binary directory.

I'm OK with renaming this if you think that it doesn't express what it actually is.

And this reminds me again of writing up a good doc on the build system of quasar ...

Reply:
OK, thanks - the explanation makes things clearer (and good naming is usually a pretty subjective thing anyway). How about 1 of the following...
```IS_CMAKE_BUILD_ENTRY_POINT```

Or, with inverted logic
```IS_QUASAR_PY_BUILD```

Reply:
Actually I really like the 1st one; sounds really meaningful.
Will adapt it.
Would you have any other remarks ?

Reply:
Nope. Looks good.